src/lib/consensus/proof_of_stake.ml

```diff
@@ -41,1433 +50,1381 @@ let uint32_of_int64 x = x |> Int64.to_int64 |> UInt32.of_int64
 
 let int64_of_uint32 x = x |> UInt32.to_int64 |> Int64.of_int64
 
-module Make (Inputs : Intf.Proof_of_stake.Inputs.S) : Intf.S = struct
-  module Time = Inputs.Time
+module Constants = struct
+  include Constants
 
-  module Constants = struct
-    include Inputs.Constants
+  module Slot = struct
+    let duration = Constants.block_window_duration
 
-    module Slot = struct
-      let duration = Time.Span.of_ms block_duration_ms
+    let duration_ms = Int64.of_int block_window_duration_ms
+  end
 
-      let duration_ms = block_duration_ms
-    end
+  module Epoch = struct
+    (** Number of slots =24k in ouroboros praos *)
+    let size = UInt32.of_int (3 * c * k)
 
-    module Epoch = struct
-      (** Number of slots =24k in ouroboros praos *)
-      let size = UInt32.of_int (3 * c * k)
+    (** Amount of time in total for an epoch *)
+    let duration =
+      Time.Span.of_ms Int64.Infix.(Slot.duration_ms * int64_of_uint32 size)
+  end
 
-      (** Amount of time in total for an epoch *)
-      let duration =
-        Time.Span.of_ms Int64.Infix.(Slot.duration_ms * int64_of_uint32 size)
-    end
+  (** The duration of delta *)
+  let delta_duration =
+    Time.Span.of_ms (Int64.of_int (Int64.to_int Slot.duration_ms * delta))
+end
 
-    (** The duration of delta *)
-    let delta_duration =
-      Time.Span.of_ms (Int64.of_int (Int64.to_int Slot.duration_ms * delta))
-  end
+module Proposal_data = struct
+  type t = {stake_proof: Coda_base.Stake_proof.t; vrf_result: Sha256.Digest.t}
+  [@@deriving bin_io]
 
-  module Proposal_data = struct
-    type t = {stake_proof: Coda_base.Stake_proof.t; vrf_result: Sha256.Digest.t}
-    [@@deriving bin_io]
+  let prover_state {stake_proof; _} = stake_proof
+end
 
-    let prover_state {stake_proof; _} = stake_proof
-  end
+let genesis_ledger_total_currency =
+  Coda_base.Ledger.to_list Genesis_ledger.t
+  |> List.fold_left ~init:Balance.zero ~f:(fun sum account ->
+         Balance.add_amount sum
+           (Balance.to_amount @@ Coda_base.Account.balance account)
+         |> Option.value_exn ?here:None ?error:None
```

Comment:
why did you add these optional params?

Reply:
For some reason, it started generating a type error. Really uncertain as to why though. I think maybe because I converted the jbuild file into a dune file?